qt/aqt/log.py

```diff
@@ -1,148 +1,101 @@
+    ):
         super().__init__(rootnode)
-        self.path = Path(path)
+        self.loggerDict = existing_loggers
+        self.logs_path = Path(logs_path)
 
     def getLogger(self, name: str) -> logging.Logger:
+        if not name.startswith(ADDON_LOGGER_PREFIX) or name in self.loggerDict:
+            return super().getLogger(name)
+
+        # Create a new add-on logger
         logger = super().getLogger(name)
-        if name.startswith(self.TAG) and RotatingFileHandler not in [
-            handler.__class__ for handler in logger.handlers
-        ]:
-            module = name[len(self.TAG) :].partition(".")[0]
-            path = self.path / "addon" / module / f"{module}.log"
-            path.parent.mkdir(parents=True, exist_ok=True)
-
-            handler = RotatingFileHandler(path)
-            logger.addHandler(handler)
-            handler.setFormatter(FORMATTER)
-        return logger
 
-    # non standard LoggerManager method
-    def find_logger_module(self, module: str) -> logging.Logger | None:
-        return self.loggerDict.get(f"{self.TAG}{module}")
-
-    def find_logger_output(self, module: str) -> Path | None:
-        logger = self.find_logger_module(module)
-        if not logger:
-            return
-        handlers = [
-            handler
-            for handler in logger.handlers
-            if isinstance(handler, RotatingFileHandler)
-        ]
-        if not handlers:
-            return
-        return Path(handlers[0].stream.name)
-
-
-def config(path: Path | str, **kwargs) -> None:
-    """configure the main logging
-
-    path: rootdir to store the addon logs
-
-    Example:
-        import aqt.log
-        aqt.log.config(
-            pm.addonFolder(),
-            level=logging.DEBUG
+        module = name.split(ADDON_LOGGER_PREFIX)[1].partition(".")[0]
+        path = get_addon_logs_folder(self.logs_path, module=module) / f"{module}.log"
+        path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Keep the last 10 days of logs
+        handler = TimedRotatingFileHandler(
+            filename=path, when="D", interval=1, backupCount=10, encoding="utf-8"
         )
-    """
+        handler.setFormatter(FORMATTER)
 
-    # we save the logs already defined
-    old = logging.root.manager.loggerDict
+        logger.addHandler(handler)
 
-    handlers = [
-        logging.StreamHandler(stream=sys.stdout),
-    ]
-    handlers[0].setFormatter(FORMATTER)
-    logging.basicConfig(handlers=handlers, **kwargs)
-    logging.Logger.manager = LoggerManager(path, logging.root)
-    logging.root.manager.loggerDict.update(old)
+        return logger
 
-    logging.captureWarnings(True)
 
-    # silence these loggers:
-    loggers = [
-        "waitress.queue",
-    ]
-    for logger in loggers:
-        logging.getLogger(logger).setLevel(logging.CRITICAL)
-        logging.getLogger(logger).propagate = False
+def get_addon_logs_folder(logs_path: Path | str, module: str) -> Path:
```

Comment:
In a newest commit, I've placed a new funtion in qt/aqt/profiles.py `def logFolder(self) -> str:` wouldn't make more sense using it?

Reply:
I think it makes sense to pass the log folder path in, so that we can keep `log.py` as free from depending on a live profile as possible. However, it turns out I was not actually doing that properly – the code in `addons.py` was actually supplying `pm.profileFolder()` to this function rather than `pm.logFolder()`. Just fixed that in the latest commits. So  good catch, thank you!

Reply:
good to go.